#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 50
}
Viewpoint {
  orientation 0.9740408624770246 -0.04628303543147895 -0.2215903401691112 0.4224703850192778
  position -1.4222322511274719 -3.8548773776520253 8.499001494070951
  followType "None"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  rotation 1 0 0 1.57
  floorSize 5 5
  wallHeight 0.5
}
Robot {
  translation -2.25 -2.25 0.0467551
  rotation -0.0028693113008759485 -0.0023385146048412703 0.9999931491775839 0.2506746423338448
  children [
    Display {
      width 1000
      height 1000
    }
    GPS {
      translation -0.09 0.005 0.1
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Cone {
            bottomRadius 0.05
            height 0.04
          }
        }
      ]
    }
    InertialUnit {
      translation -0.09 0.00499994 0.052583
      rotation 0 0 1 1.5708
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.04
            radius 0.05
          }
        }
      ]
      name "imu"
    }
    Lidar {
      translation 0 0 0.2
      rotation 1 0 0 1.57
      children [
        Shape {
          appearance CorrugatedMetal {
          }
          geometry Box {
            size 0.05 0.05 0.08
          }
        }
      ]
      horizontalResolution 1024
      fieldOfView 6.28
      numberOfLayers 1
      minRange 0.05
      maxRange 7
    }
    DEF rightback HingeJoint {
      jointParameters HingeJointParameters {
        position 1092.1886679133074
        axis 0 1 0
        anchor 0.1 0.11 0
      }
      device [
        DEF motor4 RotationalMotor {
          name "motor4"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation 0.1 0.11 0
        rotation 0 -1 0 1.0855755359406203
        children [
          DEF wheel Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.6 0.6 1
              }
            }
            geometry Cylinder {
              height 0.05
              radius 0.05
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF rightfront HingeJoint {
      jointParameters HingeJointParameters {
        position 1105.0901981464226
        axis 0 1 0
        anchor -0.1 0.11 0
      }
      device [
        DEF motor1 RotationalMotor {
          name "motor1"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation -0.1 0.11 0
        rotation 0 -1 0 0.7504159171846116
        children [
          DEF wheel Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.6 0.6 1
              }
            }
            geometry Cylinder {
              height 0.05
              radius 0.05
            }
          }
        ]
        name "solid(2)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF leftfront HingeJoint {
      jointParameters HingeJointParameters {
        position 176.26923560580488
        axis 0 1 0
        anchor -0.1 -0.11 0
      }
      device [
        DEF motor2 RotationalMotor {
          name "motor2"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation -0.1 -0.11 0
        rotation 0 1 0 0.3400470047764584
        children [
          DEF wheel Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.6 0.6 1
              }
            }
            geometry Cylinder {
              height 0.05
              radius 0.05
            }
          }
        ]
        name "solid(3)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF leftback HingeJoint {
      jointParameters HingeJointParameters {
        position 189.16961484537524
        axis 0 1 0
        anchor 0.1 -0.11 0
      }
      device [
        DEF motor3 RotationalMotor {
          name "motor3"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation 0.1 -0.11 0
        rotation 0 1 0 0.6740556299876415
        children [
          DEF wheel Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.6 0.6 1
              }
            }
            geometry Cylinder {
              height 0.05
              radius 0.05
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF body Shape {
      appearance BlanketFabric {
      }
      geometry Box {
        size 0.3 0.16 0.08
      }
    }
  ]
  name "robot(2)"
  boundingObject USE body
  physics Physics {
  }
  controller "main"
}
Wall {
  translation 0.75 1 0
  rotation 1.6952594753969508e-09 0.707104781184338 0.7071087811831002 -3.1415853071795863
  name "wall1"
  size 3.5 0.5 0.1
}
Wall {
  translation -0.75 -1 0
  rotation 1 0 0 1.5708
  name "wall2"
  size 3.5 0.5 0.1
}
